use-cases/using-SRIOV-and-kata.md

```diff
@@ -0,0 +1,313 @@
+   $ # verify `CONFIG_VFIO_IOMMU_TYPE1`, `CONFIG_VFIO=m` and `CONFIG_VFIO_PCI=m` are set as well.
+   $ sudo make olddefconfig 
+   ```
+
+   You might want to modify the kernel `Makefile` to add a unique identifier 
+   to the `EXTRAVERSION` variable prior to running the make. Including the `EXTRAVERSION` 
+   variable causes the `uname -r` command to indicate that a customized kernel is
+   installed and running.
+
+3. Build and install the kernel:
+
+   ```
+   $ make -j3 #and go get coffee
+   $ make modules -j 3
+   $ sudo make modules_install
+   $ sudo make install
+   ```
+
+4. Edit grub to enable `intel-iommu`:
+
+   ```
+   edit /etc/default/grub and add intel_iommu=on to cmdline:
+   - GRUB_CMDLINE_LINUX=""
+   + GRUB_CMDLINE_LINUX="intel_iommu=on"
+   sudo update-grub
+   ```
+
+5. Reboot the system and verify:
+
+   Host system should be ready now. Reboot the system.
+   ```
+   sudo reboot
+   ```
+
+   To verify the kernel version and the kernel command line, take a look at
+   `/proc/version` and `/proc/cmdline`
+
+6. Verify Intel VT-d has been initialized:
+   
+   To check if Intel VT-d initialized correctly, look for the following 
+   line in the `dmesg` output:
+   ```
+   DMAR: Intel(R) Virtualization Technology for Directed I/O
+   ```
+
+   Older kernels use a different prefix (e.g. PCI-DMA)
+   ```
+   PCI-DMA: Intel(R) Virtualization Technology for Directed I/O
+   ```
+
+7. Add the `vfio-pci` module
+
+   ```
+   sudo modprobe vfio-pci
+   ```
+
+8. If necessary, add PCI quirk for SR-IOV NIC
+
+   ```
+   $ find /sys/kernel/iommu_groups/ -type l
+   ```
+   The previous command verifies that your NIC appears in its own IOMMU group
+   and no other devices appear in the same group. In the rare case where your 
+   PCI NIC does not appear in its own group, it is likely that the NIC does 
+   not support ACS or you built and ran an old kernel. Depending on your NIC 
+   and if it enforces isolation, you might resolve this by adding a 
+   `pcie_acs_override=` option to your kernel command line and reboot. 
+   See [PCIE-ACS-override-option](https://lkml.org/lkml/2013/5/30/513) for
+   detailed information about this option.
+
+## Set up the SR-IOV Device
+
+All the steps in prior sections need to be performed just once to prepare the 
+SR-IOV host systems. The following will be needed per system boot in order to
+facilitate setting up a physical device's virtual functions.
+
+The following procedure sets up your SR-IOV device and needs to be done per 
+system boot. Set up includes loading a device driver, finding out how many 
+virtual functions (VF) you can create, and creating those virtual functions.
+Once you create VFs you cannot increase or decrease the number of VFs without 
+first setting the number back to zero. Based on this, it is expected that you
+set the number of VFs for a physical device just once.
+
+1. Add `vfio-pci` device driver
```

Comment:
In order to keep this list parallel I suggest editing so we can have a colon at the end of steps one and two (steps three through six are already formatted in this way). Perhaps moving the "`vfio-pci` is a driver used to reserve a VF PCI device.  Add it:" line to follow the example and striking "Add it:" since we already say to add the driver in the heading.